### ISSUE TITLE
Deprecate SettlementCenterEnum

### ISSUE THREAD

**Issue by tomhealey-icma** [finos/common-domain-model#2213]
The SettlementBase attribute SettlementCenter is a enum of two options, EuroclearBank or ClearstreamBankingLuxembourg. The use of enums for this purpose is not necessary and actually more complicated to match settlement parties to a specific enum. Instead there should be settlement parties with various roles needed to support the settlement process. The proposal is to deprecate SettlementCenter and add settlementParty Party(0..*) and settlementRole(0..*) with the option to reference the parties on the trade.

**Comment by lolabeis** [comment]
Reopening as #2214 has been reverted by #2326

**Comment by lolabeis** [comment]
> We could use AncillaryParty that includes AncillaryRoleEnum and partyReference Party (1..*) <"Specifies the party, or parties, associated to the ancillary role.">
[metadata reference]
We would mark settlementCentre SettlementCentreEnum (0..1) for deprecation.

Thanks @tomhealey-icma. Some implementation considerations:
- `SettlementBase` should only include the `AncillaryRoleEnum` attribute, no `Party` (or reference)
- The link between `Party` and `AncillaryRoleEnum` should be done at the `TradableProduct` level using the existing `AncillaryParty` attribute
- `AncillaryRoleEnum` needs a new enum value for settlement centre
- A complete contribution for v7 should include a remapping of the old to the new attributes for existing trade samples that enables regression tests to pass, rather than just marking the old attribute as "deprecated"

**Comment by tomhealey-icma** [comment]
If we proceed in adding AncillaryParty to AssetBase then we can just add settlementPartyRole AncillaryPartyRole (0..*) since parties linked to the product and not the trade will be available in assetBase. We can then deprecate settlementCenterEnum.

**Comment by tomhealey-icma** [comment]
When splitting parties into counterparties and ancillary parties the question is, if a party is acting as an ancillary party can it also be a counterparty or would it's role change from being a ancillary party to a counterparty even though it's the same party. Examples, executing broker acting as principal, clearing firm acting as counterparty, issuer acting as buyer or seller. My initial list for PartyRoleEnum is:

BeneficialOwner
Beneficiary
Borrower
Broker
Buyer
Client
ContractualParty
CounterPartyAffiliate
CounterPartyUltimateParent
Counterparty
DisputingParty
ExecutingBroker
HedgingParty
Lender

**Comment by CDM-ReleaseManagement-OT** [comment]
Hi @tomhealey-icma, do you want to discuss this on the upcoming CRWG?

**Comment by tomhealey-icma** [comment]
The difference with this proposal and the prior is that issued assets, bonds, equities have predefined settlement centers that are defined at issuance. So it's important that settlement center is part of settlement base, thus I suggest:
settlementCentreParty Party (0..1)
settlementCentreRole PartyRole (0..1)

Ancillary is a very broad range of almost any type of institution that is not a counterparty to the transaction. Also AncillaryPartyRoleEnum doesn't include any settlement centers named previously. If the concern is that settlementbase is only in the product, then I agree and suggest that settlement is added to the trade.

**Comment by ReleaseManagement-BP** [comment]
Discussed at [CDM Contribution Review Working Group - May 12th 2026](https://github.com/finos/common-domain-model/discussions/4701)

@lolabeis  suggested using 'ancilliaryParty' instead, @tomhealey-icma suggested this was a broad range for institutions, Leo suggested not having any specified party near 'economicTerms', and an alternative should be agreed. 'settlementBase' should not having any party available as a settlement party. @lolabeis  suggests adding ancillary, and adding 'settlementCentre' under the enum for this purpose. @tomhealey-icma  will refactor based off Leo's suggestion

**Comment by tomhealey-icma** [comment]
@lolabeis : Referring to your previous comments I want to clarify the need:
SettlementBase should only include the AncillaryRoleEnum attribute, no Party (or reference)
> We could add AncillaryRoleEnum to settlement centre but none of the other AncillaryRoleEnum have anything to do with settlement, DisruptionEventsDeterminingParty, ExtraordinaryDividendsParty ...we also need to define other parties to the settlement, such as the settlement agent, custodian, bank, ...

The link between Party and AncillaryRoleEnum should be done at the TradableProduct level using the existing AncillaryParty attribute
> For securities, like bonds and equities, the settlement centre is set out in the terms of the product not on the trade. If you have defined pre-defined CDM products, you don't want to set the settlement centre on the tradable product -> ancillaryParty every time you do a trade. It also logically fits in settlement base which is already in productBase.

AncillaryRoleEnum needs a new enum value for settlement centre. 
> As stated above I think we need a different list of party types for settlement. 

A complete contribution for v7 should include a remapping of the old to the new attributes for existing trade samples that enables regression tests to pass, rather than just marking the old attribute as "deprecated"

 I'm open to considering another solution that is in the product tree and not at the trade level like tradeableProduct.

**Comment by ReleaseManagement-BP** [comment]
Discussed at [CDM Contribution Review Working Group - May 26th, 2026](https://github.com/finos/common-domain-model/discussions/4705)

@tomhealey-icma attempted to enact the solutions suggestions on the last meeting. Updates have been added on why this has not worked, and is looking for additional follow-up comments. A new PR has been submitted which @lolabeis will look to review

**Comment by lolabeis** [comment]
@tomhealey-icma The solution I propose is below.

You have 2 enums for "ancillary role":
* `AncillaryRoleEnum` (used in the `TradableProduct` tree)
* `AssetAncillaryPartyRoleEnum` (used in the `Asset` tree, from your recent work on Asset/Party)

You can add the relevant party roles related to settlement in those 2 enums. At least SettlementCentre will be needed in both, maybe add custodian and others only in the Asset-related one? This will allow you to define the settlement centre party at either the Asset or TradableProduct level, as required.

Then in `SettlementBase`, replace the `settlementCentre` attribute by 2 attributes using those 2 enums, with an `optional choice` condition to make them mutually exclusive. There should also be a condition in `SettlementBase` to enforce that the value for either of these attributes, when set, must be `SettlementCentre` (you can emulate how this is done for `AncillaryRoleEnum` in `AdditionalDisruptionEvents`).

**Comment by ReleaseManagement-BP** [comment]
Discussed at [CDM Contribution Review Working Group - June 9th, 2026](https://github.com/finos/common-domain-model/discussions/4785)

@tomhealey-icma was not available on the call to discuss this. His current PR needs to change, and @lolabeis has suggested the solution, which needs to be implemented by Monday

**Comment by tomhealey-icma** [comment]
I agree with the proposed approach and will implement the changes in the PR.

**Comment by tomhealey-icma** [comment]
My preference is to not implement a condition to require a SettlementCentre as it imposes a model restriction that is actually a state condition. As an example, when creating a new bond the settlement centre may not be known and this condition would block creating other elements of the settlementbase.

**Comment by ReleaseManagement-BP** [comment]
Discussed at [CDM Contribution Review Working Group - June 23rd, 2026](https://github.com/finos/common-domain-model/discussions/4789)

@lolabeis has reviewed the latest PR, has some comments and will leave a detailed review. This has been made an required choice opposed to a optional choice. Additional comments will follow

**Comment by lolabeis** [comment]
@tomhealey-icma 
Reflecting on our discussion at the [CDM Contribution Review Working Group - July 7th, 2026](#4877), a later evolution / sophistication of the solution could be as follows:
- Build a new complex type `SettlementCentre` that pairs up an AssetTypeEnum and an AssetAncillaryPartyRoleEnum
- Remove existing `settlementCentre` attribute and replace with new `settlementCentre SettlementCentre (0..*)`
- Remove `cashSettlementCentre` and `securitySettlementCentre` attributes, superseded by the new  multi-cardinality `settlementCentre` (effectively 2 expected, 1 for Cash and 1 for Security)

```
type SettlementCentre:
  assetType AssetTypeEnum (1..1)
  centre AssetAncillaryPartyRoleEnum (1..1)

type SettlementBase:
  ...
  settlementCentre SettlementCentre (0..*)
  ...
``` 

To ponder in a follow-on issue in the context of the CDM's next development version...

In the meantime, having `cashSettlementCentre` and `securitySettlementCentre` as enum attributes would closely approach that model - hence my suggestion to keep them.

**Comment by CDM-ReleaseManagement-AP** [comment]
@tomhealey-icma This PR is missing release notes could you please provide the release notes
